### ISSUE TITLE
Dealer Hits When Shouldn't

### ISSUE THREAD

**Issue by CatGuardian** [kedoska/engine-blackjack#45]
There is a bug in the engine where the dealer hits when they shouldn't.

Here is scenario 1:
The player stands when they have 15 points.
The dealer was dealt 16 points.
Since the dealer has less than 17, the dealer hits.

The problem is that the dealer has already won and shouldn't hit.

Here is scenario 2:
The player stands when they have 15 points.
The dealer was dealt less than 15 points.
The dealer hits and gets 16 points.
Since the dealer has less than 17, the dealer hits again.

The problem is that the dealer has already won and shouldn't hit.

**Comment by kedoska** [comment]
Hi there,

I implement the following rule: 

> The dealer must hit until the cards total 17 or more points.

You can configure the engine to match the rule in both ways: soft or hard 17.

Please provide some sources or game variation that describes your requirement.

**Comment by CatGuardian** [comment]
Ah, ok I saw that option and wasn't sure what it meant. I thought it was talking about what the dealer does with Aces. Hard 17 meaning if the dealer has an ace and the points are 17 and above then the dealer would _not_ hit. And Soft 17 meaning if the dealer has an ace and the points are 17 and above then the dealer would hit.

But that makes sense. I will try changing that setting and see if it does what I want it to.

Thanks! Great work on this engine by the way. It is very helpful! Much appreciated.

P.S. I defined some typescript definition files for your project, let me know if you want those because it would be awesome if this project had an @types npm package.
